src/app/clusters/fan-control-server/fan-control.h

```diff
@@ -0,0 +1,393 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace FanControl {
+
+class Delegate;
+
+class Instance : public AttributeAccessInterface, public CommandHandlerInterface
```

Comment:
Why does this need to use CommandHandlerInterface?

Reply:
It does not _have_ to use the `CommandHandlerInterface`. 

I have a simpler implementation on a branch that uses the `emberAfFanControlClusterStepCallback` to a very small delegate that handles the command. 

I (perhaps wrongly) perceived that we wanted to move away from that style of implementation to a more object-oriented approach where we instantiate an object per cluster and have an application delegate linked to it. 

So, for that reason, I used `CommandHandlerInterface` so it was easier to offload the command to the correct delegate.

I did this work because we had some extra time before TE2 and I (again, perhaps wrongly) perceived this as the way we wanted to move forward. 

Happy to discuss and also happy to revert to the previous approach if you would prefer, I went this direction because I thought it was desirable and FanControl had been provisional until now.

Reply:
So here are the tradeoffs: using CommandHandlerInterface in its current state forces all the code for an optional command to be included, even for a device that does not implement that command.  But it does allow the  architecture you describe, which may be what we want to move toward.

Using the static cluster callback to handle the command means the unused code can get stripped out at link time if the command is not actually supported, but then requires some sort of lookup table to find the right delegate, or requires having a single delegate across all endpoints or something.

It might be worth just documenting why we are using CommandHandlerInterface here.